src/lib/operators/jit_operator/operators/jit_read_tuples.cpp

```diff
@@ -73,18 +73,18 @@ void JitReadTuples::before_chunk(const Table& in_table, const Chunk& in_chunk, J
     const auto column_id = input_column.column_id;
     const auto segment = in_chunk.get_segment(column_id);
     const auto is_nullable = in_table.column_is_nullable(column_id);
-    resolve_data_and_segment_type(*segment, [&](auto type, auto& typed_segment) {
-      using Type = typename decltype(type)::type;
-      create_iterable_from_segment<Type>(typed_segment).with_iterators([&](auto it, auto end) {
-        using IteratorType = decltype(it);
-        if (is_nullable) {
-          context.inputs.push_back(
-              std::make_shared<JitSegmentReader<IteratorType, Type, true>>(it, input_column.tuple_value));
-        } else {
-          context.inputs.push_back(
-              std::make_shared<JitSegmentReader<IteratorType, Type, false>>(it, input_column.tuple_value));
-        }
-      });
+
+    segment_with_iterators(*segment, [&](auto it, const auto end) {
+      using Type = typename decltype(it)::ValueType;
+      using IteratorType = decltype(it);
+
+      if (is_nullable) {
```

Comment:
Not part of this PR, but I wonder how well the branch prediction works here. As this will not change for a long time, can the CPU execute this as efficiently as if the `if` was outside the loop?

Reply:
No expertise in branch prediction, but I'd expect it to have learned that "is_nullable" is either true or false after a few iterations? Should I extract the condition?

Reply:
well, it does add a few loc, but changed it